### PR TITLE
remove the demo role dependency and add the necessary variables to th…

### PR DESCRIPTION
**Pull request by arbabnazar**
@edx/devops remove the `demo` role from dependency as we don't need it and it fill the logs with bunch of skipping tasks. Thanks

FYI:

```
09:49:41 TASK: [common | restart rsyslogd] ********************************************* 
09:49:42 skipping: [xxxx.sandbox]
09:49:42 
09:49:42 TASK: [demo | create demo app and data dirs] ********************************** 
09:49:42 ok: [xxxx.sandbox]
09:49:42 
09:49:42 TASK: [demo | check out the demo course] ************************************** 
09:49:46 ok: [xxxx.sandbox]
09:49:46 
09:49:46 TASK: [demo | import demo course] ********************************************* 
09:49:46 skipping: [xxxx.sandbox]
09:49:46 
09:49:46 TASK: [demo | create some test users and enroll them in the course] *********** 
09:49:46 skipping: [xxxx.sandbox] => (item={'password': 'xxx', 'email': 'honor@example.com', 'mode': 'honor'})
09:49:46 skipping: [xxxx.sandbox] => (item={'password': 'xxx', 'email': 'audit@example.com', 'mode': 'audit'})
09:49:46 skipping: [xxxx.sandbox] => (item={'password': 'xxx', 'email': 'verified@example.com', 'mode': 'verified'})
09:49:46 
09:49:46 TASK: [demo | create staff user] ********************************************** 
09:49:46 skipping: [xxxx.sandbox]
09:49:46 
09:49:46 TASK: [demo | add test users to the certificate whitelist] ******************** 
09:49:46 skipping: [xxxx.sandbox] => (item={'password': 'xxx', 'email': 'honor@example.com', 'mode': 'honor'})
09:49:46 skipping: [xxxx.sandbox] => (item={'password': 'xxx', 'email': 'audit@example.com', 'mode': 'audit'})
09:49:46 skipping: [xxxx.sandbox] => (item={'password': 'xxx', 'email': 'verified@example.com', 'mode': 'verified'})
09:49:46 
09:49:46 TASK: [demo | seed the forums for the demo course] **************************** 
09:49:46 skipping: [xxxx.sandbox] => (item={'password': 'xxx', 'email': 'honor@example.com', 'mode': 'honor'})
09:49:46 skipping: [xxxx.sandbox] => (item={'password': 'xxx', 'email': 'audit@example.com', 'mode': 'audit'})
09:49:46 skipping: [xxxx.sandbox] => (item={'password': 'xxx', 'email': 'verified@example.com', 'mode': 'verified'})
```

### CODE REVIEW

File: playbooks/roles/testcourses/defaults/main.yml [hunk 1]
```diff
@@ -11,3 +11,19 @@
 # Defaults for role testcourses
 #
 TESTCOURSES_EXPORTS: []
+
+demo_test_users:
```
Comment (feanil) [inline playbooks/roles/testcourses/defaults/main.yml]: `demo_test_users` should be renamed if this var is being used by the `testcourses` role now.  Maybe `testcourses_app_users`

**Comment by e0d** [comment]
@benpatterson any issues with this change?

I'm :+1:

**Comment by benpatterson** [comment]
@e0d @arbabnazar is the demo course contained in TESTCOURSES_EXPORTS ? (I can't see that var.)

If so, this gets a 👍 from me....

**Comment by jibsheet** [comment]
@benpatterson 

```
TESTCOURSES_EXPORTS:
  - github_url: "https://github.com/edx/demo-performance-course.git"
    install: "{{ ENABLE_PERFORMANCE_COURSE }}"
    course_id: "course-v1:DemoX+PERF101+course"
  - github_url: "https://github.com/edx/demo-test-course.git"
    install: "{{ ENABLE_DEMO_TEST_COURSE }}"
    course_id: "course-v1:edX+Test101+course"
  - github_url: "https://github.com/edx/edx-demo-course.git"
    install: "{{ ENABLE_EDX_DEMO_COURSE }}"
    course_id: "course-v1:edX+DemoX+Demo_Course"
```

**Comment by feanil** [comment]
Is there any reason why we shouldn't delete the `demo` role all together?

**Comment by e0d** [comment]
@arbabnazar needs rebase, then lets merge.  We can answer @feanil 's question separately.

**Pull request closed without merging**